Update scoring help page to explain how lack of support for SwiftPM and WASM will affect score (#9360)
dart-lang/pub-dev: doc/help-scoring.md

    windows:
    linux:
  ```

#### Support modern toolchains (SwiftPM and Wasm)

> To ensure the ecosystem stays up-to-date, pub points are reduced for packages that do not support modern toolchains for their declared platforms:

Reviewer: ```suggestion
To ensure the ecosystem stays up-to-date, pub points are awarded for packages that support modern toolchains for their declared platforms:
```

Reviewer: it's a subtle and possibly meaningless distinction, but we try to phrase the scoring in positive terms. 

You are no penalized or punished for not doing something. you are awarded for what you do.

> * Flutter plugins that support iOS or macOS will be penalized if they do not support [Swift Package Manager](https://docs.flutter.dev/packages-and-plugins/swift-package-manager/for-plugin-authors).

Reviewer: ```suggestion
* Flutter plugins that support iOS or macOS will only receive full score if they support [Swift Package Manager](https://docs.flutter.dev/packages-and-plugins/swift-package-manager/for-plugin-authors).
```

> * Packages that support the Web platform will be penalized if they are not [WebAssembly (Wasm) ready](https://dart.dev/web/wasm).

Reviewer: ```suggestion
* Packages that support the Web platform will be awarded more points if they are [WebAssembly (Wasm) ready](https://dart.dev/web/wasm).
```
